docs/rules/no-dismantled-character-class.md

```diff
@@ -0,0 +1,64 @@
+# Disallow characters which are made with multiple code points in character class syntax (no-dismantled-character-class)
+
+Unicode includes the characters which are made with multiple code points.
+RegExp character class syntax (`/[abc]/`) cannot such a character as a character. For example, `❇️` is made by `❇` (`U+2747`) and VARIATION SELECTOR-16 (`U+FE0F`). If this character is in RegExp character class, it will match to either `❇` (`U+2747`) or VARIATION SELECTOR-16 (`U+FE0F`) rather than `❇️`.
```

Comment:
The first sentence here is a bit confusing: "...cannot such a character as a character." Maybe this should say, "...cannot directly match such a character"?

Reply:
I like what you've suggested better :smile: